### ISSUE TITLE
denali publish command fails

### ISSUE THREAD

**Issue by seawatts** [denali-js/denali#400]
```bash
denali publish
cli v0.0.17 [local]  | denali v0.0.37 [local]

✔ Addon built
✔ Tests passed
✖ Publish failed
NestedError: "publish" command failed
    at Function.<anonymous> (./seawatts/denali-jwt/node_modules/denali-cli/lib/command.ts:89:13)
    at Generator.throw (<anonymous>)
    at rejected (./seawatts/denali-jwt/node_modules/tslib/tslib.js:94:69)
    at <anonymous>
Caused By: Error: Command failed: npm publish
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! denali-jwt@1.0.0-beta.6 prepublish: `in-publish && echo "Use 'denali publish' instead." && exit 1 || not-in-publish`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the denali-jwt@1.0.0-beta.6 prepublish script.
```

**Comment by davewasmer** [comment]
This looks like an issue in your repo:

```
npm ERR! denali-jwt@1.0.0-beta.6 prepublish: `in-publish && echo "Use 'denali publish' instead." && exit 1 || not-in-publish`
```

Looks like you are using a recent denali, but still have the old prepublish hook in your package.json.

tl;dr:

- Replace the "prepublish" hook with a `"prepublishOnly": "npm run build"` and make sure your build script is `denali build`
- Remove `in-publish` from your project

Longer explanation:

The old prepublish script was placed in the addon blueprint to ensure that publishing would _always_ fail. Then the old build script, when it copied the package.json into `dist`, actually removed that hook. So basically, it was setup to block publishing from the root dir by accident, and instead force you to rely on denali publish, which would build _and then publish from dist_, as was needed under the old build system, pre `main-dir`

Now, with `main-dir`, we don't need to publish from a different folder, so the publish command changed - it no longer does the whole "remove the hook from package.json and publish from dist". It just builds and publishes - that's it.

**Comment by davewasmer** [comment]
Closing since I'm pretty sure this is an issue in the consuming project. But feel free to leave a comment if you find otherwise and I'll happily reopen

**Comment by seawatts** [comment]
Sounds great! I'll check when I get home
